Automattic/jetpack
# Assistant: Add site discount endpoints

#### Changes proposed in this Pull Request:

This PR adds an endpoint to fetch a site specific discount.

#### Jetpack product discussion



Context: pbNhbs-1xX-p2

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






Prerequisites:
- Download this branch and build the plugin
- Make sure that your Jetpack test site is talking to your sandbox
- Run the command to watch changes
- Apply D76497-code to your sandbox, and follow the diff instructions

PR instructions:
- After having followed the diff instructions, your site should now have a site coupon
- In `projects/plugins/jetpack/_inc/client/recommendations/index.jsx`, add `<QuerySiteDiscount />` to the fragment returned by the component
- Open the Network panel of your browser, and refresh the `/wp-admin/admin.php?page=jetpack#/recommendations` page
- You should see that a call was made to `/wp-json/jetpack/v4/site/discount`
- Check that the payload matches the information you see in the coupon tool (code, expiry date, discount, used or not)
